lute/task/src/task.cpp

```diff
@@ -71,16 +74,79 @@ int lua_defer(lua_State* L)
     return lua_yield(L, 0);
 };
 
+#define debugstack(L, pos) printf("lua stack spot [%d] is a \"%s\"\n", pos, lua_typename(L, lua_type(L, pos)))
+
+int lua_delay(lua_State* L)
+{
+    int type = lua_type(L, 1);
+    uint64_t milliseconds = 0;
+
+    // Handle overloads
+    switch (type)
+    {
+    default:
+        luaL_errorL(L, "invalid type %s passed into task.delay", lua_typename(L, lua_type(L, 1)));
+        break;
+
+    case LUA_TNUMBER:
+        milliseconds = static_cast<uint64_t>(lua_tonumber(L, 1) * 1000);
+        break;
+    case LUA_TUSERDATA:
+        double seconds = getSecondsFromTimespec(getTimespecFromDuration(L, 1));
+        milliseconds = static_cast<uint64_t>(seconds * 1000);
+
+        break;
+    };
```

Comment:
code style for this kinda switch, basically put default last, and group each case with line breaks around it

```suggestion
    switch (type)
    {
    case LUA_TNUMBER:
        milliseconds = static_cast<uint64_t>(lua_tonumber(L, 1) * 1000);
        break;

    case LUA_TUSERDATA:
        double seconds = getSecondsFromTimespec(getTimespecFromDuration(L, 1));
        milliseconds = static_cast<uint64_t>(seconds * 1000);
        break;

    default:
        luaL_errorL(L, "invalid type %s passed into task.delay", lua_typename(L, lua_type(L, 1)));
        break;
    };
```